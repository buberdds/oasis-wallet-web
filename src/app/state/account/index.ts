import { PayloadAction } from '@reduxjs/toolkit'
import { Transaction } from 'app/state/transaction/types'
import { createSlice } from 'utils/@reduxjs/toolkit'
import { AccountState, Account } from './types'

export const initialState: AccountState = {
  address: '',
  liquid_balance: 0,

  accountError: null,
  transactions: [],
  transactionsError: null,
  loading: true,
}

const slice = createSlice({
  name: 'account',
  initialState,
  reducers: {
    clearAccount(state, action: PayloadAction<void>) {
      Object.assign(state, initialState)
    },
    fetchAccount(state, action: PayloadAction<string>) {},
    accountLoaded(state, action: PayloadAction<Account>) {
      state.accountError = null
      Object.assign(state, action.payload)
    },
    accountError(state, action: PayloadAction<string>) {
      state.accountError = action.payload
    },
    transactionsLoaded(state, action: PayloadAction<Transaction[]>) {
      state.transactionsError = null
      state.transactions = action.payload
    },
    transactionsError(state, action: PayloadAction<string>) {
      state.transactionsError = action.payload
    },
    setLoading(state, action: PayloadAction<boolean>) {
      state.loading = action.payload
    },
  },
})

export const { actions: accountActions } = slice

export default slice.reducer
