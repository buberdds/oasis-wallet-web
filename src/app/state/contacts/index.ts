import { PayloadAction } from '@reduxjs/toolkit'
import { createSlice } from 'utils/@reduxjs/toolkit'
import { ContactsState, Contact } from './types'

export const initialState: ContactsState = {}

const slice = createSlice({
  name: 'contacts',
  initialState,
  reducers: {
    add(state, action: PayloadAction<Contact>) {
      state[action.payload.address] = action.payload
    },
    update(state, action: PayloadAction<Contact>) {
      state[action.payload.address] = action.payload
    },
    delete(state, action: PayloadAction<string>) {
      delete state[action.payload]
    },
  },
})

export const { actions: contactsActions } = slice

export default slice.reducer
