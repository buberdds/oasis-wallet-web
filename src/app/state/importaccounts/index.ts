import { PayloadAction } from '@reduxjs/toolkit'
import { ErrorPayload } from 'types/errors'
import { createSlice } from 'utils/@reduxjs/toolkit'
import { ImportAccountsListAccount, ImportAccountsState, ImportAccountsStep } from './types'

export const initialState: ImportAccountsState = {
  accounts: [],
  deviceRequested: false,
  showAccountsSelectionModal: false,
}

const slice = createSlice({
  name: 'importAccounts',
  initialState,
  reducers: {
    clear(state, action: PayloadAction<void>) {
      state.accounts = []
      state.error = undefined
      state.step = undefined
      state.showAccountsSelectionModal = false
    },
    deviceRequested(state, action: PayloadAction<void>) {
      state.deviceRequested = true
    },
    enumerateAccountsFromLedger(state, action: PayloadAction<void>) {
      state.step = undefined
      state.accounts = []
      state.showAccountsSelectionModal = true
    },
    enumerateAccountsFromMnemonic(state, action: PayloadAction<string>) {
      state.step = undefined
      state.accounts = []
      state.showAccountsSelectionModal = true
    },
    toggleAccount(state, action: PayloadAction<number>) {
      const index = action.payload
      state.accounts[index].selected = !state.accounts[index].selected
    },
    accountsListed(state, action: PayloadAction<ImportAccountsListAccount[]>) {
      state.accounts = action.payload
    },
    setStep(state, action: PayloadAction<ImportAccountsStep>) {
      state.step = action.payload
    },
    operationFailed(state, action: PayloadAction<ErrorPayload>) {
      state.error = action.payload
      state.step = undefined
    },
  },
})

export const { actions: importAccountsActions } = slice

export default slice.reducer
