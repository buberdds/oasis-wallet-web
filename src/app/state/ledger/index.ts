import { PayloadAction } from '@reduxjs/toolkit'
import { ErrorPayload } from 'types/errors'
import { createSlice } from 'utils/@reduxjs/toolkit'
import { LedgerAccount, LedgerState, LedgerStep } from './types'

export const initialState: LedgerState = { accounts: [] }

const slice = createSlice({
  name: 'ledger',
  initialState,
  reducers: {
    clear(state, action: PayloadAction<void>) {
      state.accounts = []
      state.error = undefined
    },
    enumerateAccounts(state, action: PayloadAction<void>) {
      state.step = undefined
      state.accounts = []
    },
    toggleAccount(state, action: PayloadAction<number>) {
      const index = action.payload
      state.accounts[index].selected = !state.accounts[index].selected
    },
    accountsListed(state, action: PayloadAction<LedgerAccount[]>) {
      state.accounts = action.payload
    },
    setStep(state, action: PayloadAction<LedgerStep>) {
      state.step = action.payload
    },
    operationFailed(state, action: PayloadAction<ErrorPayload>) {
      state.error = action.payload
      state.step = undefined
    },
  },
})

export const { actions: ledgerActions } = slice

export default slice.reducer
