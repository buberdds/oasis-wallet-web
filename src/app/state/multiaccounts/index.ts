import { PayloadAction } from '@reduxjs/toolkit'
import { ErrorPayload } from 'types/errors'
import { createSlice } from 'utils/@reduxjs/toolkit'
import { MultiAccountsListAccount, MultiAccountsState, MultiAccountsStep } from './types'

export const initialState: MultiAccountsState = { accounts: [] }

const slice = createSlice({
  name: 'multiAccounts',
  initialState,
  reducers: {
    clear(state, action: PayloadAction<void>) {
      state.accounts = []
      state.error = undefined
      state.mnemonic = undefined
      state.step = undefined
    },
    enumerateAccountsFromLedger(state, action: PayloadAction<void>) {
      state.step = undefined
      state.accounts = []
    },
    enumerateAccountsFromMnemonic(state, action: PayloadAction<string>) {
      state.step = undefined
      state.accounts = []
    },
    toggleAccount(state, action: PayloadAction<number>) {
      const index = action.payload
      state.accounts[index].selected = !state.accounts[index].selected
    },
    accountsListed(state, action: PayloadAction<MultiAccountsListAccount[]>) {
      state.accounts = action.payload
    },
    setMnemonic(state, action: PayloadAction<string>) {
      state.mnemonic = action.payload
    },
    setStep(state, action: PayloadAction<MultiAccountsStep>) {
      state.step = action.payload
    },
    operationFailed(state, action: PayloadAction<ErrorPayload>) {
      state.error = action.payload
      state.mnemonic = undefined
      state.step = undefined
    },
  },
})

export const { actions: multiAccountsActions } = slice

export default slice.reducer
