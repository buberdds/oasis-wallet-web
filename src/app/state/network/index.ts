import { PayloadAction } from '@reduxjs/toolkit'
import { createSlice } from 'utils/@reduxjs/toolkit'
import { NetworkState, NetworkType } from './types'

export const initialState: NetworkState = {
  ticker: '',
  chainContext: '',
  selectedNetwork:
    process.env.NODE_ENV && process.env.NODE_ENV !== 'production' && !process.env.REACT_APP_BYPASS_LOCAL
      ? 'local'
      : 'mainnet',
  epoch: 0,
  minimumStakingAmount: 0,
}

const slice = createSlice({
  name: 'network',
  initialState,
  reducers: {
    selectNetwork(state, action: PayloadAction<NetworkType>) {},
    networkSelected(state, action: PayloadAction<NetworkState>) {
      Object.assign(state, action.payload)
    },
  },
})

export const { actions: networkActions } = slice

export default slice.reducer
