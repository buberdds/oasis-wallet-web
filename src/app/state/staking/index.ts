import { PayloadAction } from '@reduxjs/toolkit'
import { createSlice } from 'utils/@reduxjs/toolkit'

import { DebondingDelegation, Delegation, StakingState, Validators, ValidatorDetails } from './types'

export const initialState: StakingState = {
  debondingDelegations: [],
  delegations: [],
  validators: null,
  updateValidatorsError: null,
  selectedValidatorDetails: null,
  selectedValidator: null,
  loading: false,
}

const slice = createSlice({
  name: 'staking',
  initialState,
  reducers: {
    setLoading(state, action: PayloadAction<boolean>) {
      state.loading = action.payload
    },
    fetchAccount(state, action: PayloadAction<string>) {},
    updateValidators(state, action: PayloadAction<Validators>) {
      state.updateValidatorsError = null
      state.validators = action.payload
    },
    updateValidatorsError(state, action: PayloadAction<{ error: string; validators: Validators }>) {
      state.updateValidatorsError = action.payload.error
      state.validators = action.payload.validators
    },
    updateDelegations(state, action: PayloadAction<Delegation[]>) {
      state.delegations = action.payload
    },
    updateDebondingDelegations(state, action: PayloadAction<DebondingDelegation[]>) {
      state.debondingDelegations = action.payload
    },
    validatorSelected(state, action: PayloadAction<string>) {
      state.selectedValidator = action.payload
      state.selectedValidatorDetails = null
    },
    validatorDeselected(state, action: PayloadAction<void>) {
      state.selectedValidator = null
      state.selectedValidatorDetails = null
    },
    updateValidatorDetails(state, action: PayloadAction<ValidatorDetails>) {
      state.selectedValidatorDetails = action.payload
    },
  },
})

export const { actions: stakingActions, reducer: stakingReducer } = slice

export default slice.reducer
