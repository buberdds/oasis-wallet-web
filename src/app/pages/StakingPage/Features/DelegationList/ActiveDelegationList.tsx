import { selectActiveDelegations } from 'app/state/staking/selectors'
import { Box, Heading } from 'grommet'
import React from 'react'
import { useTranslation } from 'react-i18next'
import { useSelector } from 'react-redux'
import { FadeIn } from 'app/components/Animations'

import { DelegationList } from '.'

export const ActiveDelegationList = () => {
  const { t } = useTranslation()
  const delegations = useSelector(selectActiveDelegations)
  return (
    <FadeIn>
      <Box pad="medium" background="background-front">
        <Heading margin="none" size="small">
          {t('delegations.activeDelegations', 'Active delegations')}
        </Heading>
        <DelegationList type="active" delegations={delegations ?? []} />
      </Box>
    </FadeIn>
  )
}
