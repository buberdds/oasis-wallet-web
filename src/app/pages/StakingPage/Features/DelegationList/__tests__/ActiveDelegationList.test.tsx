import userEvent from '@testing-library/user-event'
import { render, screen, waitFor } from '@testing-library/react'

import { Provider } from 'react-redux'
import { ActiveDelegationList } from '../ActiveDelegationList'
import { configureAppStore } from 'store/configureStore'
import { stakingActions } from 'app/state/staking'

const renderComponent = (store: any) =>
  render(
    <Provider store={store}>
      <ActiveDelegationList />
    </Provider>,
  )

describe('<ActiveDelegationList  />', () => {
  let store: ReturnType<typeof configureAppStore>

  beforeEach(() => {
    store = configureAppStore()
  })

  it('should match snapshot', () => {
    const component = renderComponent(store)
    store.dispatch(
      stakingActions.updateDelegations([
        {
          amount: '100',
          shares: '100',
          validatorAddress: 'test-validator',
          validator: {
            current_rate: 0.07,
            address: 'test-validator',
            rank: 1,
            status: 'active',
            name: 'test-validator',
          },
        },
      ]),
    )

    expect(component).toMatchSnapshot()

    // @TODO expect no rounding issues
    expect(screen.queryByText('7.000000000000001%')).toBeTruthy()
    expect(screen.queryByText('7%')).toBeFalsy()
  })

  it('should expand and display the delegation on click', async () => {
    renderComponent(store)
    store.dispatch(
      stakingActions.updateDelegations([
        {
          amount: '100',
          shares: '100',
          validatorAddress: 'oasis1qqv25adrld8jjquzxzg769689lgf9jxvwgjs8tha',
          validator: {
            address: 'oasis1qqv25adrld8jjquzxzg769689lgf9jxvwgjs8tha',
            rank: 1,
            status: 'active',
            name: 'test-validator1',
          },
        },
        { amount: '50', shares: '50', validatorAddress: 'oasis1qq2vzcvxn0js5unsch5me2xz4kr43vcasv0d5eq4' },
      ]),
    )

    let row = screen.getByText(/test-validator1/)
    expect(row).toBeVisible()
    userEvent.click(row)

    const details = await screen.findByTestId('validator-item')
    row = screen.getAllByText(/test-validator1/)[0]

    expect(details).toBeVisible()
    userEvent.click(row)
    await waitFor(() => expect(details).not.toBeVisible())
  })
})
