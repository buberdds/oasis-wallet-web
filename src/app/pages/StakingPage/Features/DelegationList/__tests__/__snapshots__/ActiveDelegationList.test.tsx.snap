// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActiveDelegationList  /> should match snapshot 1`] = `
.c11 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #666666;
  stroke: #666666;
}

.c11 g {
  fill: inherit;
  stroke: inherit;
}

.c11 *:not([stroke])[fill="none"] {
  stroke-width: 0;
}

.c11 *[stroke*="#"],
.c11 *[STROKE*="#"] {
  stroke: inherit;
  fill: none;
}

.c11 *[fill-rule],
.c11 *[FILL-RULE],
.c11 *[fill*="#"],
.c11 *[FILL*="#"] {
  fill: inherit;
  stroke: none;
}

.c17 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #00C781;
  stroke: #00C781;
}

.c17 g {
  fill: inherit;
  stroke: inherit;
}

.c17 *:not([stroke])[fill="none"] {
  stroke-width: 0;
}

.c17 *[stroke*="#"],
.c17 *[STROKE*="#"] {
  stroke: inherit;
  fill: none;
}

.c17 *[fill-rule],
.c17 *[FILL-RULE],
.c17 *[fill*="#"],
.c17 *[FILL*="#"] {
  fill: inherit;
  stroke: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  background-color: #FFFFFF;
  color: #444444;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 24px;
}

.c18 {
  margin-left: 3px;
  font-size: 18px;
  line-height: 24px;
}

.c1 {
  margin: 0px;
  font-size: 34px;
  line-height: 40px;
  max-width: 816px;
  font-weight: 600;
}

.c4 {
  position: relative;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
  max-width: 100%;
  color: false;
  background-color: false;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  text-align: left;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  width: 100%;
  background-color: false;
  min-height: 56px;
  border-bottom-width: 1px;
  border-bottom-color: #AAAAAAaa;
  border-bottom-style: solid;
}

.c7 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  font-size: 18px;
  font-weight: 500;
  color: false;
  padding-left: 16px;
  padding-right: 16px;
}

.c15 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  padding-left: 16px;
  padding-right: 16px;
  word-break: break-word;
}

.c8 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  max-width: 100%;
  min-width: 100px;
  min-width: 34px;
  max-width: 34px;
}

.c9 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  max-width: 100%;
  min-width: 100px;
}

.c12 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  max-width: 100%;
  min-width: 100px;
  min-width: 100px;
  max-width: 100px;
}

.c16 {
  font-weight: 400;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  width: 100%;
  box-sizing: border-box;
  font-size: 16x;
  color: false;
  background-color: false;
  min-height: 48px;
}

.c14:not(:last-of-type) {
  border-bottom-style: solid;
  border-bottom-width: 1px;
  border-bottom-color: #AAAAAAaa;
}

.c14:hover {
  color: false;
  background-color: #88888833;
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  border-bottom-color: false;
  outline-style: solid;
  outline-width: 1px;
  outline-color: false;
}

.c14:hover {
  cursor: pointer;
}

.c10 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  line-height: 1;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c10:focus {
  outline: none;
  color: false;
}

.c10:hover {
  color: false;
}

.c10 span.__rdt_custom_sort_icon__ i,
.c10 span.__rdt_custom_sort_icon__ svg {
  opacity: 0;
  color: inherit;
  font-size: 18px !important;
  height: 18px !important;
  width: 18px !important;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
  -webkit-transform-style: preserve-3d;
  -ms-transform-style: preserve-3d;
  transform-style: preserve-3d;
  -webkit-transition-duration: 125ms;
  transition-duration: 125ms;
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
}

.c10 span.__rdt_custom_sort_icon__.asc i,
.c10 span.__rdt_custom_sort_icon__.asc svg {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.c10:hover {
  cursor: pointer;
}

.c10:hover span,
.c10:hover span.__rdt_custom_sort_icon__ * {
  opacity: 1;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  position: relative;
  width: 100%;
  border-radius: inherit;
  overflow-x: auto;
  overflow-y: hidden;
  min-height: 0;
}

.c3 {
  position: relative;
  width: 100%;
  display: table;
}

@media only screen and (max-width:768px) {
  .c0 {
    padding: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    font-size: 26px;
    line-height: 32px;
    max-width: 624px;
  }
}

<body>
  <div>
    <div
      class="mocked-framer-motion-div"
    >
      <div
        class="c0"
      >
        <h1
          class="c1"
        >
          delegations.activeDelegations
        </h1>
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            <div
              class="c4 rdt_Table"
              role="table"
            >
              <div
                class="c5 rdt_TableHead"
                role="rowgroup"
              >
                <div
                  class="c6 rdt_TableHeadRow"
                  role="row"
                >
                  <div
                    class="c7 c8 rdt_TableCol"
                  />
                  <div
                    class="c7 c8 rdt_TableCol"
                  />
                  <div
                    class="c7 c9 rdt_TableCol"
                  >
                    <div
                      class="c10 rdt_TableCol_Sortable"
                      id="column-name"
                      role="columnheader"
                      tabindex="0"
                    >
                      <div>
                        validator.name
                      </div>
                      <span
                        class="asc __rdt_custom_sort_icon__"
                      >
                        <svg
                          aria-label="Down"
                          class="c11"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="m2 8.35 10.173 9.823L21.997 8"
                            fill="none"
                            stroke="#000"
                            stroke-width="2"
                          />
                        </svg>
                      </span>
                    </div>
                  </div>
                  <div
                    class="c7 c9 rdt_TableCol"
                  >
                    <div
                      class="c10 rdt_TableCol_Sortable"
                      id="column-amount"
                      role="columnheader"
                      tabindex="0"
                    >
                      <div>
                        delegations.delegatedAmount
                      </div>
                      <span
                        class="asc __rdt_custom_sort_icon__"
                      >
                        <svg
                          aria-label="Down"
                          class="c11"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="m2 8.35 10.173 9.823L21.997 8"
                            fill="none"
                            stroke="#000"
                            stroke-width="2"
                          />
                        </svg>
                      </span>
                    </div>
                  </div>
                  <div
                    class="c7 c12 rdt_TableCol"
                  >
                    <div
                      class="c10 rdt_TableCol_Sortable"
                      id="column-fee"
                      role="columnheader"
                      tabindex="0"
                    >
                      <div>
                        validator.fee
                      </div>
                      <span
                        class="asc __rdt_custom_sort_icon__"
                      >
                        <svg
                          aria-label="Down"
                          class="c11"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="m2 8.35 10.173 9.823L21.997 8"
                            fill="none"
                            stroke="#000"
                            stroke-width="2"
                          />
                        </svg>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="c13 rdt_TableBody"
                offset="250px"
                role="rowgroup"
              >
                <div
                  class="c14 rdt_TableRow"
                  id="row-test-validator"
                  role="row"
                >
                  <div
                    class="c15 c8 c16 rdt_TableCell"
                    data-tag="allowRowEvents"
                    id="cell-icon-test-validator"
                    role="gridcell"
                  />
                  <div
                    class="c15 c8 c16 rdt_TableCell"
                    data-tag="allowRowEvents"
                    id="cell-status-test-validator"
                    role="gridcell"
                  >
                    <svg
                      aria-label="Status is okay"
                      class="c17"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 22c5.523 0 10-4.477 10-10S17.523 2 12 2 2 6.477 2 12s4.477 10 10 10zM7 12l4 3 5-7"
                        fill="none"
                        stroke="#000"
                        stroke-width="2"
                      />
                    </svg>
                  </div>
                  <div
                    class="c15 c9 c16 rdt_TableCell"
                    data-tag="allowRowEvents"
                    id="cell-name-test-validator"
                    role="gridcell"
                  >
                    test-validator
                  </div>
                  <div
                    class="c15 c9 c16 rdt_TableCell"
                    data-tag="allowRowEvents"
                    id="cell-amount-test-validator"
                    role="gridcell"
                  >
                    <span>
                      0.0000001
                      <span
                        class="c18"
                      />
                    </span>
                  </div>
                  <div
                    class="c15 c12 c16 rdt_TableCell"
                    data-tag="allowRowEvents"
                    id="cell-fee-test-validator"
                    role="gridcell"
                  >
                    7%
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
